nomad/structs/workload_id.go

```diff
@@ -260,6 +308,9 @@ type WorkloadIdentity struct {
 	// TTL is used to determine the expiration of the credentials created for
 	// this identity (eg the JWT "exp" claim).
 	TTL time.Duration
+
+	// Note: ExtraClaims is available on config/WorkloadIdentity but not
+	// available here on jobspecs
```

Comment:
I was about to ask if this could neatly solve #19438 and noticed your comment over there.

I like shipping this PR as-is. My main concern was that if there was a strong reason *not* to expose the field here, perhaps for security reasons, we should call those out loudly.

Reply:
Extended the comment a bit more to expand on that reasoning.